lib/cdo/legacy_varnish_helpers.rb

```diff
@@ -0,0 +1,76 @@
+# Various Ruby helper methods, originally from the `cdo-varnish` cookbook.
+#
+# Originally, we needed some complex logic to accommodate the use of Varnish
+# alongside several other caching and delivery layers in our system. We no
+# longer rely on Varnish, but we do rely on some of that old logic in some
+# places in the codebase.
+#
+# This module contains the subset of that logic that we still depend on.
+module LegacyVarnishHelpers
```

Comment:
Admittedly I can't think of a better name, but is there a way to move towards not having the word Varnish here at all? If these helpers still have a long life ahead of them, is there a different name that doesn't evoke the days of yore?

Reply:
I'll put a version of this in a comment to make it more clear, but I don't know that these helpers _do_ still have a long life ahead of them, and I would expect that this module in particular does not. In particular, it's not at all clear to me how much of this functionality we actually still need in a world without Varnish; do we still need to validate these paths now that they don't need to be compatible with multiple different layers? Can the validation logic be simplified? Which parts actually do care about those headers, now that Varnish doesn't exist?

My goal here was to rearrange some code while preserving existing behavior, but I'm not familiar enough with this part of our stack to say whether or not that behavior *should* be preserved. So, I want to make it very clear that this module represents old, legacy code which should not be added to and which ideally will be either deprecated or integrated more carefully and mindfully into other, more-permanent libraries.